event/engine.go

```diff
@@ -0,0 +1,17 @@
+package event
+
+// EngineEventType type to describe engine events
+// Engine events doesn't need the same validation that normal events need
+// These events are used internally only and cannot be emitted by services
+type EngineEventType string
```

Comment:
I don't like the fact type is prefixed with the project name, I prefer to use `InternalEvent` prefix.

The engine shouldn't limit names for a service event.  These internal events should be published on internal pubsub and shouldn't be mixed with events from services. 

A flag could be added to `Event` + method `IsInternal` rather than use magic event names

Reply:
I agree that the using prefix Engine is bad.
We did the same before with Core and we end up calling everything CoreXXX.

Internal is also not the best as we would like to create a bunch of pre-defined events that also Services could emit (eg: logs or health check).

What about `SystemEvents`?